test/fluree/db/query/negation_test.clj

```diff
@@ -11,40 +11,106 @@
                  {"ex" "http://example.com/"}]
         db0     (fluree/db ledger)
         db1     @(fluree/stage db0 {"@context" context
-                                    "insert" [{"@id" "ex:alice"
-                                               "@type" "ex:Person"
-                                               "ex:nickname" "Ali"
-                                               "ex:givenName" "Alice"
-                                               "ex:familyName" "Smith"}
-                                              {"@id" "ex:bob"
-                                               "ex:givenName" "Bob"
-                                               "ex:familyName" "Jones"}
-                                              {"@id" "ex:carol"
-                                               "ex:givenName" "Carol"
-                                               "ex:familyName" "Smith"}]})]
+                                    "insert"   [{"@id"           "ex:alice"
+                                                 "@type"         "ex:Person"
+                                                 "ex:nickname"   "Ali"
+                                                 "ex:givenName"  "Alice"
+                                                 "ex:familyName" "Smith"}
+                                                {"@id"           "ex:bob"
+                                                 "ex:givenName"  "Bob"
+                                                 "ex:familyName" "Jones"}
+                                                {"@id"           "ex:carol"
+                                                 "ex:givenName"  "Carol"
+                                                 "ex:familyName" "Smith"}]})]
+    (testing "exists"
+      (is (= [["ex:alice"]]
+             @(fluree/query db1 {"@context" context
+                                 "select"   ["?person"]
+                                 "where"    [{"@id" "?person" "@type" {"@id" "ex:Person"}}
+                                             ["exists" [{"@id" "?person" "ex:givenName" "?name"}]]]}))
+          "returns ex:Person who has an ex:givenName")
+      (is (= []
+             @(fluree/query db1 {"@context" context
+                                 "select"   ["?person"]
+                                 "where"    [{"@id" "?person" "@type" {"@id" "ex:Person"}}
+                                             ["exists" [{"@id" "?person" "ex:name" "?name"}]]]}))
+          "returns ex:Person who has an ex:name (none)"))
```

Comment:
I think this annotation is confusing because nothing is returned, but the "returns ex:Person ..." suggests at first glance that something is.

Here's how I would structure this test if I were writing it:

```clojure
(deftest negation
  (testing "Queries with negation"
    (testing "on an existing ledger containing data about people"
      (let [conn      @(fluree/connect {:method :memory})
            ledger    @(fluree/create conn "negation-test")
            context   ["https://flur.ee"
                       test-utils/default-str-context
                       {"ex" "http://example.com/"}]
            db0       (fluree/db ledger)
            db-people @(fluree/stage db0 {"@context" context
                                          "insert"   [{"@id"           "ex:alice"
                                                       "@type"         "ex:Person"
                                                       "ex:nickname"   "Ali"
                                                       "ex:givenName"  "Alice"
                                                       "ex:familyName" "Smith"}
                                                      {"@id"           "ex:bob"
                                                       "ex:givenName"  "Bob"
                                                       "ex:familyName" "Jones"}
                                                      {"@id"           "ex:carol"
                                                       "ex:givenName"  "Carol"
                                                       "ex:familyName" "Smith"}]})]
        (testing "checking for the existence of a combination of properties"
          (testing "when the combination is present in the data"
            (is (= [["ex:alice"]]
                   @(fluree/query db-people {"@context" context
                                             "select"   ["?person"]
                                             "where"    [{"@id" "?person" "@type" {"@id" "ex:Person"}}
                                                         ["exists" [{"@id" "?person" "ex:givenName" "?name"}]]]}))
                "returns the subjects with that property combination"))
          (testing "when the combination is absent from the data"
            (is (= []
                   @(fluree/query db-people {"@context" context
                                             "select"   ["?person"]
                                             "where"    [{"@id" "?person" "@type" {"@id" "ex:Person"}}
                                                         ["exists" [{"@id" "?person" "ex:name" "?name"}]]]}))
                "does not return any subjects")))))))
```

Every part of that test describes the intention of the code inside, and on any failures a full English sentence is printed out describing what was supposed to happen after the setup code.